be/src/exec/vectorized/join_hash_map.tpp

```diff
@@ -1399,12 +1399,55 @@ void JoinHashMap<PT, BuildFunc, ProbeFunc>::_probe_from_ht_for_left_anti_join_wi
     for (; i < probe_row_count; i++) {
         size_t build_index = _probe_state->next[i];
         if (build_index == 0) {
-            _probe_state->probe_index[match_count] = i;
-            _probe_state->build_index[match_count] = 0;
-            match_count++;
-
-            RETURN_IF_CHUNK_FULL()
+            if (_table_items->join_type == TJoinOp::NULL_AWARE_LEFT_ANTI_JOIN) {
+                if (_probe_state->null_array != nullptr && (*_probe_state->null_array)[i] == 1) {
+                    // when probe value is null needs match all rows in right table
+                    for (size_t j = 1; j < _table_items->row_count + 1; j++) {
+                        _probe_state->probe_index[match_count] = i;
+                        _probe_state->build_index[match_count] = j;
+                        _probe_state->probe_match_index[i]++;
+                        match_count++;
+                        _probe_state->cur_row_match_count++;
+                        RETURN_IF_CHUNK_FULL()
+                    }
+                } else {
+                    // match all null value rows in right table
+                    auto* nullable_column = ColumnHelper::as_raw_column<NullableColumn>(_table_items->key_columns[0]);
```

Comment:
`_table_items->key_columns[0]` may be non-nullable. We should firstly judge `_table_items->key_columns[0]->is_nullable()`?

Reply:
yeh, you are right.